client/python/lib/test/test_tensors.py

```diff
@@ -77,154 +77,270 @@ def test_make_ndarray_invalid_type():
+    tensor_proto = make_tensor_proto(values = values, shape=[3], dtype = DataType.DT_INT16)
     np_values = np.frombuffer(np.array(values).tobytes(), dtype = np.int16).tolist()
     assert tensor_proto.int_val == np_values
     assert tensor_proto.dtype == DataType.DT_INT16
     assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
 
-def test_make_tensor_proto_valid_string_to_float_dtype():
-    values = bytes([0x13, 0x00, 0x00, 0x00, 0x08, 0x00])
-    with pytest.raises(ValueError) as exception_info:
-        make_tensor_proto(shape= None, dtype = DataType.DT_FLOAT, values = values)
-        exception = exception_info.value
-        assert str(exception) == "buffer size must be a multiple of element size"
-
-def test_make_tensor_proto_valid_string_batch_size_2():
-    values = bytes([0x13, 0x00, 0x00, 0x00, 0x08])
-    tensor_proto = make_tensor_proto(shape=None, dtype = DataType.DT_STRING, values = [values, values])
-    np_values = np.array(values)
-    assert tensor_proto.string_val == [np_values.tobytes(), np_values.tobytes()]
-    assert tensor_proto.dtype == DataType.DT_STRING
-    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=2)])
-
-def test_make_tensor_proto_invalid_string_2D_array():
-    values = bytes([0x13, 0x00, 0x00, 0x00, 0x08, 0x00])
-    with pytest.raises(ValueError) as exception_info:
-        make_tensor_proto(shape= None, dtype = DataType.DT_STRING, values = [[values, values],[values, values]])
-        exception = exception_info.value
-        assert str(exception) == "Bytes values with dtype DT_STRING cannot have shape (2, 2)"
-
-def test_make_tensor_proto_valid_2_dims_shape():
-    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
-    tensor_proto = make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
-    np_values = np.array(values, np.float32)
-    assert tensor_proto.tensor_content == np_values.tobytes()
-    assert tensor_proto.dtype == DataType.DT_FLOAT
-    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=2), TensorShapeProto.Dim(size=3)])
+@pytest.mark.parametrize("params", [
+    {"values" : [bytes([0x13, 0x00, 0x00, 0x00, 0x08]) ,bytes([0x13, 0x00, 0x00, 0x00, 0x08])]},
+    {"values" : [bytes([0x13, 0x00, 0x00, 0x00, 0x08]) ,bytes([0x13, 0x00, 0x00, 0x00, 0x08])], "shape" : [2]},
+    {"values" : [bytes([0x13, 0x00, 0x00, 0x00, 0x08]) ,bytes([0x13, 0x00, 0x00, 0x00, 0x08])], "dtype" : DataType.DT_STRING},
+    {"values" : [bytes([0x13, 0x00, 0x00, 0x00, 0x08]) ,bytes([0x13, 0x00, 0x00, 0x00, 0x08])], "dtype" : DataType.DT_INT8},
+    {"values" : [bytes([0x13, 0x00, 0x00, 0x00, 0x08]) ,bytes([0x13, 0x00, 0x00, 0x00, 0x08])], "shape" : [10], "dtype" : DataType.DT_INT8},
+    {"values" : [bytes([0x13, 0x00, 0x00, 0x00, 0x08]) ,bytes([0x13, 0x00, 0x00, 0x00, 0x08])], "shape" : [2], "dtype" : DataType.DT_STRING},
+])
+def test_make_tensor_proto_valid_string_batch_size_2(params):
```

Comment:
We get a lot of if..else... blocks here. Is it possible to remove them? For example by creating a mapping (condition -> expected_value) instead of adding additional comparisons.